client/components/ProductsList.js

```diff
@@ -1,22 +1,42 @@
-import { ResourceItem, ResourceList, TextStyle } from "@shopify/polaris";
+import { ResourceList, TextStyle, Stack, Thumbnail } from "@shopify/polaris";
 import React from "react";
 
 export function ProductsList({ data }) {
   return (
-    <ResourceList
+    <ResourceList // Defines your resource list component
       showHeader
-      resourceName={{ singular: "product", plural: "products" }}
-      items={data}
-      renderItem={({ id, title }) => {
+      resourceName={{ singular: "Product", plural: "Products" }}
+      items={data.nodes}
+      renderItem={(item) => {
+        const media = (
+          <Thumbnail
+            source={
+              item.images.edges[0] ? item.images.edges[0].node.originalSrc : ""
```

Comment:
Rethinking this, we should probably make `media = null` if `item.images.edges[0]` is empty.

Reply:
hmmmm i think having an empty thumbnail looks slightly better because everythings aligned
![image](https://user-images.githubusercontent.com/30154753/140166207-840a6d22-4687-4bd8-8ad9-666d04bc5ec1.png) vs 
![image](https://user-images.githubusercontent.com/30154753/140172661-6ca082f3-0f2f-4b5b-9405-b2cb11ef8025.png)